Library/src/Multiplayer/SpaceEntity.cpp

```diff
@@ -805,6 +805,8 @@ void SpaceEntity::DeserialiseFromPatch(IEntityDeserialiser& Deserialiser)
 		Deserialiser.LeaveComponents();
 	}
 
+	EntitySystem->ResolveRootHierarchy(this);
```

Comment:
This made me realize I was missing a few things:
- CreateObject was missing an optional parent ptr (This was in the TDD)
- Parent reolution now happens in the local object patch
- Also updated to resolbe when adding the pending entity

Reply:
Ive updated this to make it less error-prone and less confusing + tested it works correctly in the following circumstances:

- Retrieve objects when entering space
- Locally creating objects
- Receiving objects
- object patch

I also created a test which allowed me to create an object from another connection and ensured its sent and updated correctly. This test still exists behind a macro as it will probably be useful in the future.